### DISCUSSION TITLE
BYOT - bring-your-own-transport - BIO pair vs 2 x mem BIOs?

### DISCUSSION THREAD

**Discussion by space88man** [openssl/openssl#23418] [Q&A] [answered]
For bring-your-own-transport apps (those that want to integrate into another event loop with its own socket handling) what are the pros and cons of using a BIO pair or 2 x mem BIOs?

Most current examples seem to be gravitating to the BIO pair pattern but I would like to understand when you might use 2 x mem BIOs.

**Comment by t8m** [comment, marked as answer]
Given the BIO pair is basically just 2 mem BIOs crosslinked, it really depends on what you find more convenient for your use-case.
